src/components/DataTable.ts

```diff
@@ -47,131 +50,154 @@ import './datatable.scss';
     DataTableFooter,
     Visibility
   ],
-  host: {
```

Comment:
Why is it more ideal to do it the other way? Just curious, this seems a bit cleaner given the implementation detail.